doc/users/fonts.rst

```diff
@@ -0,0 +1,113 @@
+All about Matplotlib and Fonts!
+===============================
+
+The story of fonts has been quite eventful throughout time. It involves
+contributions of tech giants such as the likes of Adobe, Apple and Microsoft.
+
+Types
+-----
+In practice, there are 3 types Matplotlib supports (in addition to
+'core fonts', more about which is explained later in the guide):
+
+.. list-table:: Type of Fonts
+   :header-rows: 1
+
+   * - Type 1 (PDF)
+     - Type 3 (PDF/PS)
+     - TrueType (PDF)
+   * - One of the oldest types, introduced by Adobe
+     - Similar to Type 1 in terms of introduction
+     - Newer than previous types, used commonly today, introduced by Apple
+   * - Restricted subset of PostScript, charstrings are in bytecode
+     - Full PostScript language, allows embedding arbitrary code
+       (in theory, even render fractals when rasterizing!)
+     - Include a virtual machine that can execute code!
+   * - These fonts support font hinting
+     - Do not support font hinting
+     - Hinting supported (virtual machine processes the "hints")
+   * - Difficult to subset!
+     - Easy to subset!
+     - Very hard to subset!
```

Comment:
I'm not sure this is "very hard" to subset, it's just a lot of work: it needs someone to understand the font format, build the implementation and test it. There are libraries like ttconv and fonttools that can do it, so it makes sense to use them. Likewise with Type 1, but neither of those two libraries helps us with it. Software like pdftex produces subsets of Type 1 fonts, but I don't think the TeX license is compatible with ours so we can't just extract the code from pdftex.

Reply:
Reworded the points 👍🏼